### PR TITLE
Yet another configuration file PR

### PR DESCRIPTION
**Pull request by jmcph4**
## Issue Addressed

#2748 

## Proposed Changes

See #3079.

## Additional Info

This PR is purely a revival of #3079.

### CODE REVIEW

File: common/clap_utils/src/preprocess.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            return Err(ClapPreprocessingError::Yaml(serde_yaml::Error::custom(
+                "Unable to parse YAML table",
+            )))
+        }
+    };
+    Ok(string_value)
+}
+
+/// This function and `expand_args_from` are inspired by `argmatches::expand_args_from` and
+/// `argmatches::expand_args_from` respectively, but differ in the following ways:
```
Comment (realbigsean) [inline common/clap_utils/src/preprocess.rs]:
```suggestion
/// This function and `expand_args_from` are inspired by `argfile::expand_args_from` and
/// `argfile::expand_args_from` respectively, but differ in the following ways:
```

File: common/clap_utils/src/preprocess.rs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+///
+/// - This has no recursion.
+/// - `parser` must return a `Result<Vec<OsString>, ClapPreprocessingError>` as opposed to a `Vec<Argument>`.
+/// - This tracks and passes along to `parser` all cli flags provided by the user.
+/// - This searches for config files based on a provided flag name rather than a prefix.
+pub fn expand_args<F>(
+    parser: F,
+    flag_name: &str,
+) -> Result<Vec<std::ffi::OsString>, ClapPreprocessingError>
```
Comment (realbigsean) [inline common/clap_utils/src/preprocess.rs]: `OsString` is also imported so we can delete the path prefix (there's a few of these in this file)

```suggestion
) -> Result<Vec<OsString>, ClapPreprocessingError>
```

File: common/clap_utils/src/preprocess.rs [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        Ok(toml) => to_arguments(cli_flags, toml, toml_value_to_string)?,
+        Err(_) => {
+            let parsed = serde_yaml::from_str(content)?;
+            to_arguments(cli_flags, parsed, yaml_value_to_string)?
+        }
+    };
+    Ok(data)
+}
+
+/// This converts a givent map of config into a `Vec<OsString>`, for consumption by
```
Comment (realbigsean) [inline common/clap_utils/src/preprocess.rs]:
```suggestion
/// This converts a given map of config into a `Vec<OsString>`, for consumption by
```

File: book/src/file-config.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+Would be equivalent to this YAML config:
+```bash
+$ lighthouse beacon_node --config-file ./beacon-config.yaml
+```
+```yaml
+debug-level: "debug"
+port: 8000
+http-port: 6052
+staking: true
+eth1-endpoints: ["http://localhost:8545", "http://localhost:9545"]
```
Comment (realbigsean) [inline book/src/file-config.md:27]: These examples are a little out of date, this should probably be updated to `execution-endpoint` and only take a single value. Although it is nice to have an array example, so maybe we should add `validator-monitor-pubkeys` instead

Comment (realbigsean) [reply book/src/file-config.md:27]: we should also maybe note in the docs that arrays can also work as a single CSV string `validator-monitor-pubkeys: "0x123..456,0x456..123"`

**Review by realbigsean — CHANGES_REQUESTED** [review]
Thanks for picking this up!

**Comment by michaelsproul** [comment]
Lets revisit this for 4.7.0

**Comment by dapplion** [comment]
We are slowly adopting clap v4 with derive syntax, which would allow us to finally have proper file config :)

Closing this approach in favor of the former

**Pull request closed without merging**